apps/clearance_ui/src/components/PurlDetails.tsx

```diff
@@ -18,23 +18,25 @@ type Props = {
 
 const PurlDetails = ({ purl }: Props) => {
     const parsedPurl = parsePurlAndQualifiers(purl);
-    const packageName =
+    const mainPurl =
+        "pkg:" +
+        (parsedPurl.type ? parsedPurl.type + "/" : "") +
         (parsedPurl.namespace ? parsedPurl.namespace + "/" : "") +
         parsedPurl.name +
-        "@" +
-        parsedPurl.version +
-        (parsedPurl.subpath ? "/" + parsedPurl.subpath : "");
+        (parsedPurl.version ? "@" + parsedPurl.version : "");
 
     return (
         <Accordion type="single" collapsible>
             <AccordionItem value="item-1">
                 <AccordionTrigger>
-                    <Badge className="rounded-md">{packageName}</Badge>
+                    <Badge className="break-all rounded-md px-1">
+                        {mainPurl}
+                    </Badge>
                 </AccordionTrigger>
                 <AccordionContent className="text-xs">
-                    {parsedPurl.type && (
+                    {parsedPurl.subpath && (
```

Comment:
As briefly mentioned on Slack, my proposal is to make a `subpath` go last, after the  `qualifiers`, as the subpath only makes sense if the `qualifiers` point to a repository.